daemons/controld/controld_messages.c

```diff
@@ -439,6 +440,18 @@ relay_message(xmlNode * msg, gboolean originated_locally)
         }
     }
 
+    // If the DC is not yet selected
+    if ((strcmp(task, CRM_OP_PING) == 0) && (controld_globals.dc_name == NULL)) {
+        if (is_for_dc) {
+            xmlNode *reply = create_ping_reply(msg);
+            sys_to = crm_element_value(reply, PCMK__XA_CRM_SYS_TO);
```

Comment:
No need, sys_to is already set

Reply:
Initially, the `sys_to` contains the `"dc"` (to where the `msg` is addressed). And here we change it to the hexadecimal node address.

Reply:
Ah thanks, of course. I missed that this uses `reply` instead of `msg`.